be/src/util/block_compression.cpp

```diff
@@ -370,27 +388,39 @@ class ZlibBlockCompression : public BlockCompressionCodec {
 };
 
 Status get_block_compression_codec(segment_v2::CompressionTypePB type,
-                                   const BlockCompressionCodec** codec) {
+                                   std::unique_ptr<BlockCompressionCodec>& codec) {
+    BlockCompressionCodec* ptr = nullptr;
     switch (type) {
     case segment_v2::CompressionTypePB::NO_COMPRESSION:
-        *codec = nullptr;
-        break;
+        codec.reset(nullptr);
+        return Status::OK();
     case segment_v2::CompressionTypePB::SNAPPY:
-        *codec = SnappyBlockCompression::instance();
+        ptr = new SnappyBlockCompression();
```

Comment:
could write this way:  codec.reset(new SnappyBlockCompression());

Reply:
Yes, it's an easier way to use codec.reset(new SnappyBlockCompression()). 

But I changed to the current more complex way, new a ptr first and then call ptr->init() and call codec.reset(ptr) finally, to avoid reset codec to a not properly initialized prt.